### PR TITLE
add Synology enhanced app

### PR DESCRIPTION
**Pull request by damda58**
image

### CODE REVIEW

**Comment by Subtixx** [comment]
Your config.blade.php contains issues.
```php
{!! Form::text('config[override_url]',(isset($item) ? $item->getconfig()->override_url : null), array('placeholder' => __('app.apps.override'), 'id' => 'override_url', 'class' => 'form-control')) !!}
```

need to be replaced with:
```php
 {!! Form::text('config[override_url]', (isset($item) && isset($item->getconfig()->override_url) ? $item->getconfig()->override_url : null), array('placeholder' => __('app.apps.override'), 'id' => 'override_url', 'class' => 'form-control')) !!}
```

File: Synology/Synology.php [hunk 1]
```diff
@@ -1,5 +1,68 @@
...
+	
+	 public function login()
+    {
+        $password = $this->config->password;
+		$user = $this->config->username;
+		
+        $attrs = [
+			'cookies' => $this->jar,			
+        ];
+        return parent::appTest($this->url('/webapi/auth.cgi?api=SYNO.API.Auth&version=2&method=Login&account='.$user.'&passwd='.$password.'&session=SurveillanceStation&format=cookie'),$attrs);
```
Comment (Subtixx) [inline Synology/Synology.php:22]: Should probably be execute not appTest

File: Synology/config.blade.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<h2>{{ __('app.apps.config') }} ({{ __('app.optional') }}) @include('items.enable')</h2>
+<div class="items">
+    <div class="input">
+        <label>{{ strtoupper(__('app.url')) }}</label>
+        {!! Form::text('config[override_url]',(isset($item) ? $item->getconfig()->override_url : null), array('placeholder' => __('app.apps.override'), 'id' => 'override_url', 'class' => 'form-control')) !!}
+    </div>
+	    <div class="input">
+        <label>{{ __('app.apps.username') }}</label>
+        {!! Form::text('config[username]', (isset($item) ? $item->getconfig()->username : null), array('placeholder' => __('app.apps.username'), 'data-config' => 'username', 'class' => 'form-control config-item')) !!}
```
Comment (Subtixx) [inline Synology/config.blade.php:9]: You need to check if `$item->getconfig()->username` with `isset` first.

File: Synology/config.blade.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<h2>{{ __('app.apps.config') }} ({{ __('app.optional') }}) @include('items.enable')</h2>
+<div class="items">
+    <div class="input">
+        <label>{{ strtoupper(__('app.url')) }}</label>
+        {!! Form::text('config[override_url]',(isset($item) ? $item->getconfig()->override_url : null), array('placeholder' => __('app.apps.override'), 'id' => 'override_url', 'class' => 'form-control')) !!}
```
Comment (Subtixx) [inline Synology/config.blade.php:5]: You need to check if `$item->getconfig()->override_url` with `isset` first.

**Comment by Subtixx** [comment]
Also there are way more changes than just synology. I'd suggest making a new PR with the fixes implemented and also with only the synology enhanced app.

**Comment by KodeStar** [comment]
What @Subtixx said ^^

**Pull request closed without merging**